ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -187,6 +189,11 @@ data instance BlockQuery (ShelleyBlock proto era) :: Type -> Type where
                     Map (SL.KeyHash 'SL.StakePool (EraCrypto era))
                         (SL.RewardInfoPool))
 
+  GetPoolState
+    :: Maybe (Set (SL.KeyHash 'SL.StakePool (EraCrypto era)))
+    -> BlockQuery (ShelleyBlock proto era)
+                  (SL.PState (Era.Crypto era))
```

Comment:
@JaredCorduan @lehins Is `PState` part of the "public API"? IE something that we want to expose in an "official" query like this new `GetPoolState`?